contracts/MultiPartyEscrow.sol

```diff
@@ -93,23 +95,53 @@ contract MultiPartyEscrow {
         require(balances[msg.sender] >= value, "Insufficient balance in the contract.");
         require(signer != address(0));
 
+        require(_openChannel(msg.sender, signer, recipient, groupId, value, expiration), "Unable to open channel");
+        return true;
+    }
+    
+    //open a channel on behalf of the user. Sender should send the signed permission to open the channel
+    function openChannelByThirdParty(address sender, address signer, address recipient, bytes32 groupId, uint256 value, uint256 expiration, uint256 messageNonce, uint8 v, bytes32 r, bytes32 s) 
+    public
+    returns(bool) 
+    {
+        require(balances[sender] >= value, "Insufficient balance");
+
+        require(messageNonce >= block.number-5 && messageNonce <= block.number+5, "Invalid message nonce");
+
+        //compose the message which was signed
+        bytes32 message = prefixed(keccak256(abi.encodePacked(this, msg.sender, signer, recipient, groupId, value, expiration, messageNonce)));
```

Comment:
I would propose to add prefix "__openChannelByThirdParty" to the message  (```abi.encodePacked("__openChannelByThirdParty",this, msg.sender, signer, recipient, groupId, value, expiration, messageNonce)```) in order to prevent all possible cross-replay attack (using this signature something else)

Reply:
@astroseger I can definitely add the prefix but it is very low probability that this signatures can be used in some other place.

Reply:
Yes. It is a question how paranoiac are we. I would prefer to add prefix, especially if it is gas free for the users. I also do not see any scenario in which such message could be reused (making test net with the same MPE address and force user to sign something there is probably impossible), but such prefix would remove any doubts (and since it is free why not add it).